### PR TITLE
[14.0][MIG] hr_expense_petty_cash

### PR DESCRIPTION
**Pull request by ps-tubtim**
Duplicate https://github.com/OCA/hr-expense/pull/22 and apply https://github.com/OCA/hr/pull/924.

Fix in case of expenses with taxes.

**Before**
Selection_080

**After**
Selection_081

### CODE REVIEW

**Comment by Saran440** [comment]
@ps-tubtim you should cherry pick from https://github.com/OCA/hr-expense/pull/22. For keep old history from @modnoting

File: hr_expense_petty_cash/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        states={"draft": [("readonly", False)]},
+    )
+
+    def action_post(self):
+        self._check_petty_cash_amount()
+        return super().action_post()
+
+    @api.constrains("invoice_line_ids", "line_ids")
+    def _check_petty_cash_amount(self):
+        petty_cash_env = self.env["petty.cash"]
```
Comment (newtratip) [inline hr_expense_petty_cash/models/account_move.py]:
```suggestion
        petty_cash_env = self.env["petty.cash"].sudo()
```
Please add sudo() on petty.cash model. This function is affect with another module such as purchase module.

I have to create vendor bill on purchase order by procurement dept (Not be in accounting group), it's raise permission error.
Selection_222

Comment (ps-tubtim) [reply hr_expense_petty_cash/models/account_move.py]: Thank you for your suggestion. Updated.

**Comment by Saran440** [comment]
@bluemix you can review petty cash here.

**Review by Saran440 — APPROVED** [review]
Code Reviewed. 👍

**Comment by hitrosol** [comment]
I would like to test it in the runboat, but I can't see it.

Updated: 
I'm able to test this on my local, LGTM.

**Review by hitrosol — APPROVED** [review]

**Pull request merged**